pkg/controller/gcppubsub/pubsub_wrapper.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcppubsub
+
+import (
+	"context"
+
+	"cloud.google.com/go/pubsub"
+)
+
+// This file exists so that we can unit test failures with the PubSub client.
+
+// pubSubClientCreator creates a pubSubClient.
+type pubSubClientCreator func(ctx context.Context, googleCloudProject string) (pubSubClient, error)
+
+// pubSubClient is the set of methods we use on pubsub.Client.
+type pubSubClient interface {
+	SubscriptionInProject(id, projectId string) pubSubSubscription
+	CreateSubscription(ctx context.Context, id string, cfg pubsub.SubscriptionConfig) (pubSubSubscription, error)
+	Topic(name string) *pubsub.Topic
+}
+
+var _ pubSubClient = &realGcpPubSubClient{}
```

Comment:
it feels strange to make an internal interface unless this is only for testing. if so, please comment?

Reply:
This _is_ specifically for testing (see comment at [line 25](https://github.com/knative/eventing-sources/pull/39/files#diff-3ff25a4f3284ce29b0494df3d71bd148R25)). I think the comment starting on [line 39](https://github.com/knative/eventing-sources/pull/39/files#diff-3ff25a4f3284ce29b0494df3d71bd148R39) explains it pretty well, but maybe should be expanded or moved if it isn't clear.

Reply:
Added additional comments.